### PR TITLE
OAuth: Do not enforce auto assign org role

### PR DESCRIPTION
**Pull request by Jguer**
**What this PR does / why we need it**:

- The user on first login (if no role mapping is done) will be assigned `auto assign org role` but once that's done he won't be synced if strict mode is disabled and he has no role assigned.

### CODE REVIEW

File: pkg/login/social/azuread_oauth.go [hunk 1]
```diff
@@ -68,10 +68,11 @@ func (s *SocialAzureAD) UserInfo(client *http.Client, token *oauth2.Token) (*Bas
 		return nil, ErrEmailNotFound
 	}
 
-	role, grafanaAdmin := claims.extractRoleAndAdmin(s.autoAssignOrgRole, s.roleAttributeStrict)
-	if role == "" {
+	role, grafanaAdmin := claims.extractRoleAndAdmin()
```
Comment (IevaVasiljeva) [inline pkg/login/social/azuread_oauth.go:71]: nit: I think this is the only method which was using Azure's `autoAssignOrgRole` parameter - it should be fine to remove it altogether now.

**Review by IevaVasiljeva — COMMENTED** [review]
The changes in the PR look fine to me!

I was trying to look through user creation code, and it looks like the role specified in `autoAssignOrgRole` is assigned to user ([here](https://github.com/grafana/grafana/blob/main/pkg/services/user/userimpl/user.go#L149-L155) and [here](https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/user.go#L166-L172)). I just hope that there aren't other user creation flows that we've overlooked 😬

**Review by kalleep — APPROVED** [review]
LGTM

**Comment by Jguer** [comment]
Putting back to draft as the direction we want to take this has changed

**Comment by daniellee** [comment]
@Jguer should this be in the milestone for 9.2.0 given that it wasn't merged?

**Comment by gamab** [comment]
I removed the milestone, thanks @daniellee

**Pull request closed without merging**